src-tauri/src/config.rs

```diff
@@ -0,0 +1,158 @@
+use serde::{Serialize, Deserialize};
+use home::{home_dir};
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use core::fmt::Display;
+
+const DEFAULT_CONFIG_FILENAME: &str = ".noters.yaml";
+
+pub struct Config {
+    filename: Box<Path>,
+    config_file: ConfigFile
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigFile {
+    meta: MetaInfo,
+    values: ConfigValues
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct MetaInfo {
+    version: u32
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigValues {
+    base_dir: String
+}
+
+fn get_home_dir() -> PathBuf {
+    home_dir().unwrap_or(PathBuf::from("."))
+}
+
+struct ConfigError {
```

Comment:
@fersaru , @hmlampe : Maybe we should move this into a separate module and rename it to `NoteRsError`. It is very generic and compatible to almost every Error struct, e.g. std::io:: error or serde error.

Reply:
If it is this generic and compatible to other common error struct, why implement one?

Reply:
It was the other way around. In the `Config::from_file` function, I need to use `fs::read` to read the config file and `serde_yaml` to parse it. The function should return a `Result`, since there is a command error handling in `Config::from_default_file` (if it fails, construct a `Config` with default values).  Unfortunately, the error types of `std::io` and `serde_yaml` are not compatible to each other, therefore I introduced `ConfigError` as a common error type.

This seems to be a common practice in the rust universe (see [stackoverflow](https://stackoverflow.com/questions/48430836/rust-proper-error-handling-auto-convert-from-one-error-type-to-another-with-que)).